reconcile/utils/secret_reader.py

```diff
@@ -27,8 +28,198 @@ class SupportsSecret(Protocol):
     q_format: Optional[str]
 
 
-class SecretReader:
-    """Read secrets from either Vault or a config file."""
+class TypedSecretReader(ABC):
```

Comment:
I'd really like to avoid sprinkling the code with references to "typed". This feels like an implementation detail that nobody should care about. Likewise for the `typed_read_*` methods can we just come up with some other name that more accurately describes it?

Reply:
Renamed those now.

`TypedSecretReader` -> `SecretReaderBase`
`typed_read_*` -> `read_secret`, `read_all_secret`